includes/functions.php

```diff
@@ -1561,16 +1561,20 @@ function is_self_ping( $id ) {
  * @return boolean|int The ID of the outbox item or false on failure.
  */
 function add_to_outbox( $data, $type = 'Create', $user_id = 0, $content_visibility = ACTIVITYPUB_CONTENT_VISIBILITY_PUBLIC ) {
-	$transformer = Transformer_Factory::get_transformer( $data );
+	if ( $data instanceof \Activitypub\Activity\Base_Object ) {
+		$activity = $data;
+	} else {
+		$transformer = Transformer_Factory::get_transformer( $data );
 
-	if ( ! $transformer || is_wp_error( $transformer ) ) {
-		return false;
-	}
+		if ( ! $transformer || is_wp_error( $transformer ) ) {
+			return false;
+		}
 
-	$activity = $transformer->to_object();
+		$activity = $transformer->to_object();
 
-	if ( ! $activity || is_wp_error( $activity ) ) {
-		return false;
+		if ( ! $activity || is_wp_error( $activity ) ) {
+			return false;
+		}
```

Comment:
@Automattic/fediverse Heads up! ^ This accounts for `Actors` already being "Activities" I guess and failing transformation.

Reply:
I fixed that in my Dispatcher PR, so that this should work properly... I would keep the transformation, because of possible additions to the `cc` and `to` fields.

Reply:
https://github.com/Automattic/wordpress-activitypub/pull/1186/commits/a90e03a6608975aff518318c8162753944e12663

Reply:
That doesn't seem to work. The transformer is looking at [the class itself](https://github.com/Automattic/wordpress-activitypub/blob/a90e03a6608975aff518318c8162753944e12663/includes/transformer/class-factory.php#L31) and not the instance, so it doesn't account for inheritance.

Reply:
but we should fix it in the transformer, to stay consistent!

Reply:
Fixed in 904848b1e95efcc13c067aa1bad2aff284d5d21a